pkg/formatted/param.go

```diff
@@ -0,0 +1,62 @@
+// Copyright © 2020 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package formatted
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+)
+
+func Param(p []v1beta1.Param, paramSpec []v1beta1.ParamSpec) string {
+	if len(p) == 0 {
+		return "---"
+	}
+	var str string
+	for i, param := range p {
+		if param.Value.Type == "string" {
+			paramValue := param.Value.StringVal
+
+			// Check for param type if value is not defined
+			if strings.ContainsAny(paramValue, "$") {
+				replacer := strings.NewReplacer("$", "", "(", "", ")", "", "params.", "")
+				paramName := replacer.Replace(param.Value.StringVal)
+				for _, spec := range paramSpec {
+					if spec.Name == paramName {
+						paramValue = string(spec.Type)
```

Comment:
Please add a comment over here explaining why the value of the param is being set as the type of the same